src/lib/currency/intf.ml

```diff
@@ -42,14 +42,28 @@ module type Basic = sig
 
   val to_formatted_string : t -> string
 
-  val of_int : int -> t
-
-  val to_int : t -> int
-
   val to_uint64 : t -> uint64
 
   val of_uint64 : uint64 -> t
 
+  val mina_unsafe : int -> t
```

Comment:
what is the failure behavior of the `unsafe` functions? Can that be described in comments?

Reply:
They underflow when provided with a negative int. `mina_unsafe` and `centimina_unsafe` can additionally overflow. I'll add comments.

Reply:
Can we have them return a `t Or_error.t`, to enforce safety? In that case, we won't need `unsafety` in the names.

Reply:
Often, especially in tests we use small constants or numbers drawn at random from a well-defined ranges that are certain not to overflow. In these cases `Or_error.t` or `option` that I used for the safe variants complicate the code unnecessarily. For this reason I think it's convenient to have both safe and unsafe variants.

Reply:
Moreover, changing the type of these functions would require much more substantial changes to many various places. I'd rather avoid that and keep things simple; just notify users with the function's name that it might be unwise to use it.

Reply:
Done in [fa1e867](https://github.com/MinaProtocol/mina/pull/11835/commits/fa1e8677e19893f51b5df90ec984210415506328).

Reply:
I don't think it's a good idea to allow silent failures.

The overflows can be checked at run-time, and an exception raised, if you don't want to change the function signature. In that case, the function names can be given a `_exn` suffix.